src/Plugin/resource/ResourceEntity.php

```diff
@@ -53,7 +52,12 @@ public function dataProviderFactory() {
       $field_definitions_array = $this->viewModeFields($plugin_definition['dataProvider']['viewMode']);
       $field_definitions = ResourceFieldCollection::factory($field_definitions_array);
     }
-    return new DataProviderEntity($this->getRequest(), $field_definitions, $this->getAccount(), $this->getPath(), $plugin_definition['dataProvider']);
+    $class_name = '\Drupal\restful\Plugin\resource\DataProvider\DataProviderEntity';
+    // TODO: Make this logic below alterable by the implementor user via info hook, or something similar.
+    if ($this->getEntityType() == 'taxonomy_term') {
```

Comment:
Don't we just need a ResourceTaxonomyTerm, that implements the correct ``dataProviderFactory``?